src/IonTimestamp.ts

```diff
@@ -348,6 +348,27 @@ export class Timestamp {
     return n;
   }
 
+  equals(expected : Timestamp) : boolean {
+    return this.getPrecision() === expected.getPrecision() && this.offset === expected.offset && this.instantEquals(expected);
+  }
+  instantEquals(expected : Timestamp) : boolean {
```

Comment:
This doesn't take into account timestamps with different local offsets that represent the same instant. For example,

`2007-01-02T01:00+01:00` and
`2007-01-01T23:00-01:00`

It also won't ignore decimal precision, e.g.

`2007-01-01T00:00:00.00-00:00 ` and
`2007-01-01T00:00:00.0000000000-00:00`

because it will use IonDecimal's (data model) equals implementation.

Reply:
https://github.com/amzn/ion-js/issues/132